### PR TITLE
fix(slice-machine-ui): allow clicking on the "Login required" text in the environment switcher

### PR DESCRIPTION
**Pull request by angeloashmore**
## Context

DT-1837

## The Solution

Use `@prismicio/editor-ui`'s `<InvisibleButton>` to render the button. Clicking the button opens the log in modal.

## Impact / Dependencies

None

## Checklist before requesting a review

- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/src/components/SideNav/SideNavEnvironmentSelector.tsx [hunk 1]
```diff
@@ -49,17 +50,20 @@ export const SideNavEnvironmentSelector: FC<SideNavEnvironmentSelectorProps> = (
           />
         )}
       </Box>
-      <Box flexGrow={1} flexDirection="column" overflow="hidden">
+      <Box
+        flexGrow={1}
+        flexDirection="column"
+        overflow="hidden"
```
Comment (bapmrl) [inline packages/slice-machine/src/components/SideNav/SideNavEnvironmentSelector.tsx:57]: ❓ Why is there `overflow="hidden"`?

**Review by bapmrl — APPROVED** [review]
The code looks good, but I don't have time to run it locally. I trust you checked your work against Figma 🙂

File: packages/slice-machine/src/components/SideNav/SideNavEnvironmentSelector.tsx (hunk 1 above)
Comment (angeloashmore) [reply packages/slice-machine/src/components/SideNav/SideNavEnvironmentSelector.tsx:57]: It is needed to handle text overflow when repository names are too long for the sidebar:
image

Without `overflow: hidden` on the parent:
image

**Comment by angeloashmore** [comment]
> The code looks good, but I don't have time to run it locally. I trust you checked your work against Figma 🙂

Thanks! Yes, I did. This PR actually fixes something I missed in the original implementation: that "Login required" was a button component, not just purple text.

**Review by bapmrl — APPROVED** [review]

**Pull request merged**